v3/src/components/axis/hooks/use-sub-axis.ts

```diff
@@ -331,11 +331,12 @@ export const useSubAxis = ({
     }
   }, [dataConfig, updateDomainAndRenderSubAxis])
 
-  // update d3 scale and axis when layout/range changes
+  // Render when axis length or number of sub-axes changes
   useEffect(() => {
     const disposer = reaction(
       () => {
-        return layout.getAxisLength(axisPlace)
+        return [layout.getAxisLength(axisPlace),
+          layout.getAxisMultiScale(axisPlace)?.repetitions]
```

Comment:
When returning a non-primitive like an array or object, I would add `equals: comparer.structural` to the options below. There's a subtle argument about whether or not it's strictly necessary, but I think the code is clearer that way.